legate/tester/test_system.py

```diff
@@ -82,6 +93,7 @@ def run(
         *,
         env: EnvDict | None = None,
         cwd: str | None = None,
+        timeout: int | None = 10,
```

Comment:
Doesn't really matter, since we always set this explicitly, but maybe set the default value to `None`, just so it doesn't clash with the default behavior (no default timeout) higher up in the stack?